From da118d48a4169d37c13243a6bd4126b3a1d24c72 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Sat, 31 Aug 2024 19:38:00 -0700
Subject: [PATCH 01/14] NOTCBL: REPORTED: Revert "fbdev: offb: replace
 of_node_put with __free(device_node)"

This reverts commit 96ee5c5712efb7ec5af66e3d600d6258cece0ea5.

Link: https://lore.kernel.org/4f4ac35e-e31c-4f67-b809-a5de4d4b273a@roeck-us.net/
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/video/fbdev/offb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/offb.c b/drivers/video/fbdev/offb.c
index 6f0a9851b092..91001990e351 100644
--- a/drivers/video/fbdev/offb.c
+++ b/drivers/video/fbdev/offb.c
@@ -355,7 +355,7 @@ static void offb_init_palette_hacks(struct fb_info *info, struct device_node *dp
 			par->cmap_type = cmap_gxt2000;
 	} else if (of_node_name_prefix(dp, "vga,Display-")) {
 		/* Look for AVIVO initialized by SLOF */
-		struct device_node *pciparent __free(device_node) = of_get_parent(dp);
+		struct device_node *pciparent = of_get_parent(dp);
 		const u32 *vid, *did;
 		vid = of_get_property(pciparent, "vendor-id", NULL);
 		did = of_get_property(pciparent, "device-id", NULL);
@@ -367,6 +367,7 @@ static void offb_init_palette_hacks(struct fb_info *info, struct device_node *dp
 			if (par->cmap_adr)
 				par->cmap_type = cmap_avivo;
 		}
+		of_node_put(pciparent);
 	} else if (dp && of_device_is_compatible(dp, "qemu,std-vga")) {
 #ifdef __BIG_ENDIAN
 		const __be32 io_of_addr[3] = { 0x01000000, 0x0, 0x0 };

base-commit: 311d8503ef9fa25932825c5342de7213738aad8e
-- 
2.46.0

